src/Microsoft.DotNet.Interactive/InstallPackagesMessage.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Interactive.Formatting;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Html;
+using static Microsoft.DotNet.Interactive.Formatting.PocketViewTags;
+
+namespace Microsoft.DotNet.Interactive
+{
+    // Data bag containing the data to be displayed when reporting nuget resolve progress 
+    public class InstallPackagesMessage
+    {
+        static InstallPackagesMessage ()
+        {
+            Formatter.Register<InstallPackagesMessage>(m => m.FormatAsPlainText(), PlainTextFormatter.MimeType);
+            Formatter.Register<InstallPackagesMessage>(m => m.FormatAsHtml(), HtmlFormatter.MimeType);
+            Formatter.SetPreferredMimeTypeFor(typeof(InstallPackagesMessage), PlainTextFormatter.MimeType);
+        }
+
+        public IEnumerable<string> RestoreSources;
+        public IEnumerable<string> InstalledPackages;
+        public IEnumerable<string> InstallingPackages;
+        public int Progress;
+
+        public InstallPackagesMessage(
+                IEnumerable<string> restoreSources,
+                IEnumerable<string> installedPackages,
+                IEnumerable<string> installingPackages,
+                int progress)
+        {
+            RestoreSources = restoreSources;
+            InstalledPackages = installedPackages;
+            InstallingPackages = installingPackages;
+            Progress = progress;
+        }
+
+        IHtmlContent InstallMessage(string message, IEnumerable<string> items, string progress = null)
+        {
+            if (items.Count() > 0)
+            {
+                return div(
+                    strong(message),
+                    ul(items.Select(s => li(span(s + progress)))));
+            }
+            else
+            {
+                return div();
+            }
+        }
+
+        private string FormatAsHtml()
```

Comment:
Does this not get double-encoded?

In order to avoid that, this should return `IHtmlContent`.

Reply:
If I understand this test correctly, I think it works as you would expect:   
https://github.com/dotnet/interactive/blob/54c98c456b7afeba45c599cba8aae4134a97c81e/src/Microsoft.DotNet.Interactive.Formatting.Tests/PocketViewWithFormatterTests.cs#L25

Reply:
Also just a habit: if you have a method that returns `string` but that `string` contains HTML, it's just a matter of time before it gets double encoded. But yeah this works like you expect:

![image](https://user-images.githubusercontent.com/547415/118135419-d9d60100-b3b7-11eb-899b-84d8d81d0573.png)

Reply:
If I don't have to register formatters, then this can go.